src/openms/source/FORMAT/MSPTSEFile.cpp

```diff
@@ -0,0 +1,152 @@
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Douglas McCloskey, Pasquale Domenico Colaianni $
+// $Authors: Douglas McCloskey, Pasquale Domenico Colaianni $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/FORMAT/MSPTSEFile.h>
+#include <boost/regex.hpp>
+#include <fstream>
+#include <OpenMS/CONCEPT/LogStream.h>
+
+namespace OpenMS
+{
+  MSPTSEFile::MSPTSEFile(const String& filename, MSExperiment& experiment)
+  {
+    load(filename, experiment);
+  }
+
+  void MSPTSEFile::load(const String& filename, MSExperiment& experiment) const
+  {
+    std::ifstream ifs(filename, std::ifstream::in);
+    if (!ifs.is_open())
+    {
+      throw Exception::FileNotFound(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, filename);
+    }
+    const Size BUFSIZE { 65536 };
+    char line[BUFSIZE];
+    experiment.clear(true);
+    MSSpectrum spectrum;
+    bool adding_spectrum { false }; // to avoid calling `.addSpectrum()` on empty/invalid spectra
+    boost::cmatch m;
+    boost::regex re_name("^Name: (.+)", boost::regex::no_mod_s);
+    boost::regex re_synon("^Synon: (.+)", boost::regex::no_mod_s);
+    boost::regex re_formula("^Formula: (.+)", boost::regex::no_mod_s);
+    boost::regex re_mw("^MW: (.+)", boost::regex::no_mod_s);
+    boost::regex re_cas_nist("^CAS#: (.+);  NIST#: (.+)", boost::regex::no_mod_s);
+    boost::regex re_db("^DB#: (.+)", boost::regex::no_mod_s);
+    boost::regex re_comments("^Comments: (.+)", boost::regex::no_mod_s);
+    // boost::regex re_num_peaks("^Num Peaks: .+", boost::regex::no_mod_s);
+    boost::regex re_points_line("^(?:\\d+ \\d+; )+$", boost::regex::no_mod_s);
+    boost::regex re_point("(\\d+) (\\d+); ", boost::regex::no_mod_s);
+    while (!ifs.eof())
+    {
+      ifs.getline(line, BUFSIZE);
+      if (boost::regex_search(line, m, re_name))
+      {
+        LOG_DEBUG << std::endl << std::endl << "re_name ";
+        spectrum.clear(true);
+        spectrum.setName( std::string(m[1]) );
+        LOG_DEBUG << spectrum.getName() << std::endl;
+        MSSpectrum::StringDataArrays& SDAs = spectrum.getStringDataArrays();
+        SDAs.resize(7);
+        // TODO: part of this info could be saved as integer instead of string
+        SDAs[0].setName("Synon");
```

Comment:
I am not sure if we can generalize these names to all .msp files for metabolomics.  I would imagine for instance that the "NIST#" is specific only to the NIST library.  I would also imagine that they could vary quite dramatically between different spectral library databases.  I will double check on the header names for some other prominent databases in the field (e.g., Metlin, MSCloud, etc,.) to see which fields are required and which are optional.